lpcompiler.pas

```diff
@@ -1638,9 +1647,9 @@ function TLapeCompiler.ParseMethod(FuncForwards: TLapeFuncForwards; FuncHeader:
     end;
 
     try
-      if (Tokenizer.Tok = tk_kw_Overload) or (FuncHeader.isOperator and (Tokenizer.Tok <> tk_kw_Override)) then
+      if (Tokenizer.Tok = tk_kw_Overload) or ((FuncHeader.MethodDef in [mdOperator, mdProperty]) and (Tokenizer.Tok <> tk_kw_Override)) then
       begin
-        if not FuncHeader.isOperator then
+        if not(FuncHeader.MethodDef in [mdOperator, mdProperty]) then
```

Comment:
I agree, but currently the parser will accept `overload` to be appended as long as it isn't followed by a semicolon. Might be better to not allow it at all.